andrei-s/src/main/java/org/java/mentorship/andrei_s/service/SongService.java

```diff
@@ -0,0 +1,74 @@
+package org.java.mentorship.andrei_s.service;
+
+import org.java.mentorship.andrei_s.domain.Song;
+import org.springframework.stereotype.Service;
+
+import java.util.LinkedList;
+
+@Service
+public class SongService {
+
+    private final LinkedList<Song> songs;
+    private int currentId = 0;
+
+    SongService()
+    {
+        this.songs = new LinkedList<>();
+
+        this.createNewSong(new Song(1, "Pop", 10, 10));
+        this.createNewSong(new Song(2, "Pop", 10, 10));
+        this.createNewSong(new Song(3, "Rock", 10, 11));
+        this.createNewSong(new Song(4, "Rock", 10, 11));
+        this.createNewSong(new Song(5, "Rock", 10, 12));
+    }
+
+    public Song getSongById(int songId)
+    {
+        for (Song song : songs)
```

Comment:
Generally we use Java 8 Streams where applicable.

I know we haven't discussed them (mostly due to lack of time), but have a look at this article
https://www.baeldung.com/java-8-streams

In this case, this would transform to 
return songs.stream().filter(song -> song.getId() == songId).findFirst().orElse(null);